app.py

```diff
@@ -1,2 +1,57 @@
 from flask import Flask, request
 import json
+
+
+app = Flask(__name__)
+with open("restaurants.json", "r") as read_file:
+    data = json.load(read_file)
+
+restaurants = data['restaurants']
+
+
+@app.route('/api/restaurants', methods=['GET', 'POST'])
+def all_restaurants():
+    if request.method == 'POST':
+        content = request.get_json(silent=True)
+        print(type(content))
+        if content is not None:
+            print(content)
+            restaurants.append(content['restaurant_name'])
+        response = app.response_class(
+            response=json.dumps({'restaurants': restaurants}),
+            status=201,
+            mimetype='application/json'
+        )
+        return response
+    else:
```

Comment:
Imagine that you are reading this code for the first time. you see handler like:
```python
if request.method == 'POST':
   ...
else:
   ...
```
so you think: "well, what does this `else` mean?", you look above and see `methods=['GET', 'POST']` "ah, hm, this  endpoint supports 2 operations, first one was `POST`, then another is `GET`"

then you added one more HTTP method: `methods=['GET', 'POST', 'DELETE']`
and there is:
```python
if request.method == 'POST':
   ...
elif request.method == 'DELETE':
   ...
else:
  ...
```

I guess you see what I mean.
So it is better to be explicit and write like `elif request.method == 'GET'`

OR you can use Flask's [MethodView ](https://flask.palletsprojects.com/en/1.1.x/views/#method-based-dispatching)and then there is no need in these `if-else` at all :)
(to my mind, it makes sense to use them if there are several HTTP methods for the same route)

Reply:
fixed